quickwit/quickwit-indexing/src/actors/merge_pipeline.rs

```diff
@@ -120,7 +123,11 @@ impl Actor for MergePipeline {
 impl MergePipeline {
     // TODO improve API. Maybe it could take a spawnbuilder as argument, hence removing the need
     // for a public create_mailbox / MessageCount.
-    pub fn new(params: MergePipelineParams, spawn_ctx: &SpawnContext) -> Self {
+    pub fn new(
+        params: MergePipelineParams,
+        initial_immature_splits_opt: Option<Vec<SplitMetadata>>,
```

Comment:
Is there a case where we want to call this with `None`? What is it?

Reply:
So I covered the `spawn_pipelines` branch, which is the one `apply_indexing_plan` uses, but I did not handle`spawn_pipeline`, which I think is used by the local CLI ingest. This is the reason for using an option.

I will refactor the code a little bit to eliminate the option. With some tweaking, `spawn_pipeline` can use `spawn_pipelines` under the hood.